host_applications/linux/apps/raspicam/RaspiVid.c

```diff
@@ -2020,8 +2034,29 @@ static MMAL_STATUS_T create_encoder_component(RASPIVID_STATE *state)
       }
    }
 
+   if (state->encoding == MMAL_ENCODING_H264 && state->slices > 1 && state->common_settings.width <= 1280)
+   {
+      int frame_mb_rows = VCOS_ALIGN_UP(state->common_settings.height, 16) >> 4;
+
+      if (state->slices > frame_mb_rows) //warn user if too many slices selected
+      {
+         fprintf(stderr,"H264 Slice count (%d) exceeds number of macroblock rows (%d). Setting slices to %d.\n", state->slices, frame_mb_rows, frame_mb_rows);
+         // Continue rather than abort..
+      }
+      int slice_row_mb = frame_mb_rows/state->slices;
+      if (frame_mb_rows - state->slices*slice_row_mb)
+         slice_row_mb++; //must round up to avoid extra slice if not evenly divided
```

Comment:
Answering my own question, I suspect it does as ```slice_row_mb``` will become 0, ```frame_mb_rows - 0``` is true, so we end up with ```slice_mb_row``` as 1 due to the ++.
It'd be nice to confirm this though.

Reply:
@shizukachan Can you comment on this?